code/Model/SiteTree.php

```diff
@@ -2844,6 +2871,52 @@ public function isHomePage(): bool
         return $this->URLSegment === RootURLController::get_homepage_link();
     }
 
+    public function getDefaultSearchContext()
+    {
+        return SiteTreeSearchContext::create(
+            static::class,
+            $this->scaffoldSearchFields(),
+            $this->defaultSearchFilters()
+        );
+    }
+
+    public function searchableFields()
+    {
+        $fields = parent::searchableFields();
+        // Push `FilterClass` to the top of the fields list
+        $fields = array_merge([
+            'FilterClass' => [
+                'filter' => '',
+                'general' => false,
+                'title' => _t(__CLASS__ . '.Filter_Status', 'Page status'),
+                'field' => DropdownField::class,
+            ]
+        ], $fields);
+        return $fields;
+    }
+
+    public function scaffoldSearchFields($_params = null)
+    {
+        $fields = parent::scaffoldSearchFields($_params);
+
+        // Update "Status" sources
+        $filters = CMSSiteTreeFilter::get_all_filters();
+        // Remove 'All records' as we set that to empty/default value
+        unset($filters[CMSSiteTreeFilter_Search::class]);
+        $fields->dataFieldByName('FilterClass')?->setSource($filters)->setEmptyString(CMSSiteTreeFilter_Search::singleton()->title());
+
+        // Update "Page type" sources
+        $classNames = [];
+        $validClasses = ClassInfo::getValidSubClasses(SiteTree::class);
+        $this->invokeWithExtensions('updateAllowedSubClasses', $validClasses);
+        foreach ($validClasses as $class) {
+            $classNames[$class] = SiteTree::singleton($class)->i18n_singular_name();
+        }
+        $fields->dataFieldByName('ClassName')?->setSource($classNames)->setEmptyString(_t(__CLASS__ . '.Filter_ClassName_Any', 'Any'));
```

Comment:
Split this out across multiple lines like recommended above

Reply:
Done